presto-orc/src/main/java/com/facebook/presto/orc/OrcWriter.java

```diff
@@ -717,7 +803,7 @@ private Slice toEncryptedFileStatistics(List<ColumnStatistics> statsFromRoot, in
     {
         DwrfProto.FileStatistics fileStatistics = toFileStatistics(statsFromRoot);
         DwrfDataEncryptor dwrfDataEncryptor = dwrfEncryptionInfo.getEncryptorByGroupId(groupId);
-        OrcOutputBuffer buffer = new OrcOutputBuffer(compressionParameters, Optional.of(dwrfDataEncryptor));
+        OrcOutputBuffer buffer = new OrcOutputBuffer(columnWriterOptions, Optional.of(dwrfDataEncryptor));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code contains a resource that might not have closed properly. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *buffer*. The resource is referenced at lines 807, 809, 810. The resource is closed at line 808. There are other execution paths that do not contain closure statements, for example, when DwrfProto.FileStatistics.writeTo throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).